Makefile

```diff
@@ -1,11 +1,56 @@
-BUILD_DIR:=./bin
+BUILDFLAGS ?=
+unexport GOFLAGS
 
-build: clean
-	mkdir -p ${BUILD_DIR}
-	go build -o "${BUILD_DIR}/backplane-tools" main.go
+BASE_DIR=$(shell pwd)
+BIN_DIR=${BASE_DIR}/bin
 
-clean:
-	rm -rf "${BUILD_DIR}"
+all: format mod build test lint
 
+format: vet mod fmt mockgen ci-build docs
+
+fmt:
+	@echo "gofmt"
+	@gofmt -w -s .
+	@git diff --exit-code .
+
+OS := $(shell go env GOOS | sed 's/[a-z]/\U&/')
+ARCH := $(shell go env GOARCH)
+
+.PHONY: download-goreleaser
+download-goreleaser:
+	GOBIN=${BIN_DIR} go install github.com/goreleaser/goreleaser@v1.15.0
+
+.PHONY: download-golangci-lint
+download-golangci-lint:
+	GOBIN=${BIN_DIR} go install github.com/golangci/golangci-lint/cmd/golangci-lint@v1.53.3
+
+SINGLE_TARGET ?= false
+
+# Need to use --snapshot here because the goReleaser
+# requires more git info that is provided in Prow's clone.
+# Snapshot allows the build without validation of the
+# repository itself
+.PHONY: build
+build: download-goreleaser
+	${BIN_DIR}/goreleaser build --clean --snapshot --id linux_arm64,linux_amd64,darwin_arm64,darwin_amd64 --single-target=${SINGLE_TARGET}
+
+.PHONY: release
+release:
+	${BIN_DIR}/goreleaser release --clean
```

Comment:
**thought** I'm wondering if we could have the releasing part as github workflow, that can be triggered as pipeline taking e.g. only the release version as pipeline variable. This would probably facilitate the release process by a great margin, but would still keep the process manual (start a workflow with a given version).

Reply:
I'm +1 to that long term :slightly_smiling_face: